baseapp/baseapp.go

```diff
@@ -822,7 +822,7 @@ type HasNestedMsgs interface {
 // Note, gas execution info is always returned. A reference to a Result is
 // returned if the tx does not run out of gas and if all the messages are valid
 // and execute successfully. An error is returned otherwise.
-func (app *BaseApp) runTx(mode execMode, txBytes []byte) (gInfo sdk.GasInfo, result *sdk.Result, anteEvents []abci.Event, err error) {
+func (app *BaseApp) runTx(mode execMode, txBytes []byte, tx sdk.Tx) (gInfo sdk.GasInfo, result *sdk.Result, anteEvents []abci.Event, err error) {
```

Comment:
Do we really need to provide a txBytes and optionally a tx?
What is `txBytes` doesn't correspond to the provided decoded tx?
Feels like we should have a check for that, which makes providing the tx not so useful.

Reply:
At the very least, as it prevents double decoding so it's nice to keep, we should add a comment, saying `tx` is optional, and if provided txBytes and tx must obviously correspond to the same tx.

Reply:
Can we still precise tx is optional and if provided it will be decoded from the txbytes?